snow/consensus/snowball/consensus.go

```diff
@@ -40,15 +40,19 @@ type Consensus interface {
 	Finalized() bool
 }
 
-// NnarySnowball augments NnarySnowflake with a counter that tracks the total
-// number of positive responses from a network sample.
-type NnarySnowball interface{ NnarySnowflake }
-
-// NnarySnowflake is a snowflake instance deciding between an unbounded number
-// of values. After performing a network sample of k nodes, if you have alpha
-// votes for one of the choices, you should vote for that choice. Otherwise, you
-// should reset.
-type NnarySnowflake interface {
+// ConsensusFactory produces NnarySnowflake instances
+type ConsensusFactory interface {
```

Comment:
Should we just call this `Factory`? The current naming seems pretty weird because we don't actually return a `Consensus` object anywhere in this interface.

Reply:
Ya.. I had thought of these as consensus, but that's a bit confusing because there's already a different consensus type.

I don't love making it just factory by itself, but not sure what else it would be. I'll think a bit more about this and if I don't come up with anything just make it factory.